buildSrc/src/main/groovy/airbyte-docker.gradle

```diff
@@ -0,0 +1,216 @@
+            result.add((String)imageToVersion.get(taggedImage))
+        }
+
+        return result
+    }
+
+    static String getImageHash(Project project, String taggedImage) {
+        def stdout = new ByteArrayOutputStream()
+
+        project.exec {
+            commandLine "docker", "images", "--no-trunc", "-f", "dangling=false", "--format", "{{.ID}}", taggedImage
+            standardOutput = stdout;
+        }
+
+       return "$stdout".toString().trim()
+    }
+
+    static boolean isUpToDate(Logger logger, File idFileOutput, Project project, Path dockerPath) {
+        if(idFileOutput.exists()) {
+            def taggedImage = AirbyteDockerTask.getTaggedImage(project.projectDir, 'Dockerfile')
+            logger.debug "taggedImage " + taggedImage
+
+            def current = getImageHash(project, taggedImage)
+            logger.debug "current " + current
+            def stored = ((String)project.rootProject.imageToVersion.get(taggedImage)).trim()
+            logger.debug "stored " + stored
+
+            def notUpToDate = new ArrayList<String>(getBaseTaggedImages(dockerPath.toFile())).any {baseImage ->
+                logger.debug "checking base image " + baseImage
+                def storedBase = ((String)project.rootProject.imageToVersion.get(baseImage)).trim()
+                def currentBase = getImageHash(project, baseImage)
+
+                logger.debug "storedBase " + storedBase
+                logger.debug "currentBase " + currentBase
+                if(!currentBase.equals(storedBase)) {
+                    logger.debug "did not match"
+                    return true
+                } else {
+                    logger.debug "did match"
+                    return false
+                }
+            }
+
+            if(notUpToDate) {
+                return false;
+            }
+
+            logger.debug "stored " + stored.trim()
+
+            def upToDate = current.equals(stored.trim())
+
+            logger.debug "uptodate " + upToDate.toString()
+
+            return upToDate
+        } else {
+            return false
+        }
+    }
+
+    static def createTask(Project project, String taskName, String dockerFile) {
+        if (project.file(dockerFile).exists()) {
+            def filteredProjectFiles = project.fileTree(project.projectDir).filter {
+                file -> !file.toString().contains(".venv")
+            }
+            project.task(taskName, type: AirbyteDockerTask) {
+                def dockerPath = Paths.get(project.projectDir.absolutePath, dockerFile)
+                def hash = MessageDigest.getInstance("MD5").digest(dockerPath.getBytes()).encodeHex().toString()
+                rootDir = project.rootProject.rootDir
+                projectDir = project.projectDir
+                projectFiles = filteredProjectFiles
+                idFileOutput = project.file(Paths.get(project.rootProject.rootDir.absolutePath, '.dockerversions', hash).toString())
```

Comment:
should dockerversions live in the build directory since it's a build artifact?

Reply:
I'm not positive what you're asking here. 

If you mean project-specific build directories:
Nope. It makes it harder to parse the directory for changes and we need a global way of referencing a version. If it is looking for a specific image it would have to search for the file or have a stored mapping.

If you mean root-level build directory:
Maybe? Would we want this to be cleared on a `./gradlew clean`? I'm not sure.

Reply:
i think i'd prefer not to remove it on a clean since there is already a diff computation mechanism, so this LGTM